api/buckets/service.go

```diff
@@ -1698,12 +1753,23 @@ func (s *Service) ArchiveWatch(req *pb.ArchiveWatchRequest, server pb.API_Archiv
 		return ErrArchivingFeatureDisabled
 	}
 
+	var ffsInfo *mdb.FFSInfo
+	if acct := accountFromContext(server.Context()); acct != nil {
+		ffsInfo = acct.FFSInfo
+	} else if user := userFromContext(server.Context()); user != nil {
+		ffsInfo = user.FFSInfo
+	}
+
+	if ffsInfo == nil {
+		return fmt.Errorf("no account or no FFS info associated with account")
+	}
+
 	var err error
 	ctx, cancel := context.WithCancel(server.Context())
 	defer cancel()
 	ch := make(chan string)
 	go func() {
-		err = s.Buckets.ArchiveWatch(ctx, req.GetKey(), ch)
+		err = s.Buckets.ArchiveWatch(ctx, req.GetKey(), ffsInfo.Token, ch)
```

Comment:
Making sure this is correct... in this case, if the ffs token is no longer valid, we will get the "auth token not found" error immediately and just return it a few lines below. That is fine, we don't want to do any FFS creation here.